test/test_allocation_domain.cpp

```diff
@@ -1384,4 +1384,45 @@ TEST_F(NVFuserTest, ReductionSchedulerIssue1895) {
       executor_cache.fusion(), cg_outputs, inputs, {ref}, __LINE__, __FILE__);
 }
 
+TEST_F(NVFuserTest, ReductionVectorization) {
+  auto fusion = std::make_unique<Fusion>();
+  FusionGuard fg(fusion.get());
+  long x = 2L, y = 2L, z = 2L;
+  auto tv0 = TensorViewBuilder()
+                 .ndims(3)
+                 .shape({-1, 1, -1})
+                 .contiguity({true, std::nullopt, true})
+                 .build();
+  fusion->addInput(tv0);
+  auto tv1 = TensorViewBuilder()
+                 .ndims(3)
+                 .shape({-1, -1, -1})
+                 .contiguity({true, true, true})
+                 .strideOrder({0, 1, 2})
+                 .build();
+  fusion->addInput(tv1);
+  auto s0 = IrBuilder::create<Val>(2);
+  auto tv2 = expand(tv0, {s0, s0, s0});
+  auto tv3 = mul(tv2, tv1);
+  auto tv4 = sum(tv3, {2});
+  fusion->addOutput(tv4);
+  std::vector<IterDomain*> tv4_dom = {tv4->axis(2), tv4->axis(1), tv4->axis(0)};
+  tv4->setAllocationDomain(tv4_dom, true);
+
+  // tv1 is a constant tensor, and its domains are constant.
+  // Its constant domains are used in ExactMappedExtentSubstitutionPass
+  // to substitute the domains of tv0.
+  auto options = at::TensorOptions().dtype(at::kFloat).device(at::kCUDA, 0);
+  auto t0 = at::randn({x, 1, z}, options);
+  auto t1 = at::randn({x, y, z}, options).as_strided({x, y, z}, {1, x, x * y});
+  std::vector<c10::IValue> inputs({t0, t1});
+
+  FusionExecutorCache executor_cache(std::move(fusion));
+  auto cg_outputs = executor_cache.runFusionWithInputs(inputs);
+
+  auto ref = t0.mul(t1).sum({2});
+  testValidate(
+      executor_cache.fusion(), cg_outputs, inputs, {ref}, __LINE__, __FILE__);
```

Comment:
```suggestion
  testValidate(
      executor_cache.fusion(), cg_outputs, inputs, __LINE__, __FILE__);
```

Or is there a particular reason to compute ref by hand?

Reply:
There really isn't. I just kept forgetting that when copying tests and didn't put too much thought into it during clean up.